README.md
<!-- 
This README describes the package. If you publish this package to pub.dev,
this README's contents appear on the landing page for your package.

For information about how to write a good package README, see the guide for
[writing package pages](https://dart.dev/guides/libraries/writing-package-pages). 

For general information about developing packages, see the Dart guide for
[creating packages](https://dart.dev/guides/libraries/create-library-packages)
and the Flutter guide for
[developing packages and plugins](https://flutter.dev/developing-packages). 
-->
[![BuymMeACoffee][buy_me_a_coffee_badge]][buymeacoffee]

# flutter_web_buttons

[![pub package](https://img.shields.io/pub/v/flutter_web_buttons.svg)](https://pub.dev/packages/flutter_web_buttons)

A quick way to add a growing list of customizable flat or animated buttons to your app.

## Features

* Each button features it's own constructor with plenty of options for customization.
* Includes most commonly used Social Icons
* Buttons can be used as a Hero

## Installation

In the `dependencies:` section of your `pubspec.yaml`, add the following line:

```yaml
dependencies:
  flutter_web_buttons: <latest_version>
```

In your library add the following import:

```dart
import 'package:flutter_web_buttons/flutter_web_buttons.dart';
```

## Getting Started

```dart
FlutterWebButton.textScroll(
  'Text Scroll',
  onPressed: () {},
  animationDuration: const Duration(milliseconds: 500),
  flutterWebButtonOptions: FlutterWebButtonOptions(
      buttonWidth: 200,
      buttonRadius: 30,
      textColor: Colors.pink,
      buttonBackgroundColor: Colors.transparent,
      buttonBorderColor: Colors.pink),
),

```

## Examples

| Button | Animation |
| --- | ---|
| backgroundColorChange | ![bg_color] |
| backgroundFill | ![bg_fill] |
| grow | ![button_grow] |
| iconGrow | ![icon_grow] |
| textMove | ![move_text] |
| simple | Simple button with no animation |
| socialIcon | Social icon with no animation (see list)|
| socialIconGrow | Scales a social icon to a given size |
| textColorChange | ![txt_color_change] |
| textScroll | ![txt_scroll]|
| textUnderline | ![underline] |
| buttonHighlightIconFill | ![icon_w_bg_fill] |


## Button Customization Properties

### Animation Properties

#### Some properties are not applicable for all buttons.

| Animation Specific Property | Availabilty |
| --- | -- |
| Animation Duration | All Animated Buttons |
| Animated Text Color | Button Specific |
| Animated Background Color | Button Specific |
| growAmount | Grow effect buttons only |


### Decoration Properties

| FlutterWebButtonOptions | FlutterWebIconButtonOptions |
| --- | --- |
| Height | Icon Size |
| Width | Icon Color |
| Background Color | Padding |
| Text Color | Hero Tag |
| Font Size | IsHeroWidget |
| Font Family |  |
| Padding |  |
| Eliminate Decoration |  |
| Button Radius |  |
| Border Color |  |
| Border Width |  |
| Shadow Color |  |
| Spread Radius |  |
| Blur Radius |  |
| Shadow Offset |  |
| Hero Tag |  |
| IsHeroWidget |  |



<!-- Links -->
[backgroundfill]:https://user-images.githubusercontent.com/60490869/154086649-006e9772-d0e3-455f-9403-ffad0b41a658.gif
[buy_me_a_coffee_badge]: https://img.buymeacoffee.com/button-api/?text=Buy%20Me%20A%20Coffee&emoji=&slug=djcali570&button_colour=29b6f6&font_colour=000000&font_family=Cookie&outline_colour=000000&coffee_colour=FFDD00
[buymeacoffee]:https://www.buymeacoffee.com/djcali
[bg_color]:https://user-images.githubusercontent.com/60490869/156978470-06f5eae6-b62a-4d82-a9ea-16fb0df12d46.gif
[icon_w_bg_fill]:https://user-images.githubusercontent.com/60490869/156976695-20bc5c5f-37ed-4810-900d-5d2784c6cb94.gif
[bg_fill]:https://user-images.githubusercontent.com/60490869/156979143-0ee5d402-b1d2-4911-b979-c1a7925b6f3e.gif
[button_grow]:https://user-images.githubusercontent.com/60490869/156979533-d7796263-7a54-440d-95d8-50edb5f0e1d2.gif
[icon_grow]:https://user-images.githubusercontent.com/60490869/156980305-a8f21f7e-e2b9-4bc3-a1f3-59e2f3db67c5.gif
[move_text]:https://user-images.githubusercontent.com/60490869/156981057-450e5ee2-bc7c-49ce-a644-30270df06551.gif
[txt_color_change]:https://user-images.githubusercontent.com/60490869/157051127-4e7ad83b-611b-407f-b0d7-073d6a039cf5.gif
[txt_scroll]:https://user-images.githubusercontent.com/60490869/157052261-88e09546-7647-4928-a70f-b8bce73449ea.gif
[underline]:https://user-images.githubusercontent.com/60490869/157053576-3e10dcad-d775-4104-9e30-157f34bfdeb8.gif


CHANGELOG.md
## 0.0.9

* Added widget button
* Added circle button
* Added gradient option

## 0.0.8

* Added Text Only Color Change Button

## 0.0.7

* Renamed FlutterWebIconButtonOptions to FlutterWebIconOptions to be consistant.

## 0.0.6

* Added Highlight Icon Fill button
* Added Text Move button
* Added simple icon button
* Added letterSpacing property for text
* Added option for spacing between text and the line for underline animation.
* Fixed alignment issue with underline button
* Updated example

## 0.0.5

* Added grow button
* Added icon grow button
* Added social icon grow button
* Added text underline animation
* Updated example project with new buttons

## 0.0.4

* Added code comments
* Updated example project with buttons

## 0.0.3

* Added support for using Hero widgets
* Added example project

## 0.0.2

* Fixed Github link

## 0.0.1

* Initial Release


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:flutter_web_buttons/flutter_web_buttons.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Web Buttons Example',
      theme: ThemeData(
        primarySwatch: Colors.pink,
      ),
      home: const MyHomePage(title: 'Flutter Web Buttons Demo Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  GlobalKey txtKey = GlobalKey();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Row(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                FlutterWebButton.backgroundColorChange(
                  'Background Color',
                  onPressed: () {
                    debugPrint('Button Pressed');
                  },
                  flutterWebButtonOptions: FlutterWebButtonOptions(
                      buttonWidth: 200,
                      buttonRadius: 8,
                      boxShadowColor: Colors.black.withOpacity(.8)),
                ),
                FlutterWebButton.backgroundFill('Background Fill',
                    onPressed: () {},
                    animationDuration: const Duration(milliseconds: 500),
                    flutterWebButtonOptions:
                        FlutterWebButtonOptions(buttonWidth: 200)),
                FlutterWebButton.grow(
                  'Button Grow',
                  onPressed: () {
                    debugPrint(txtKey.currentContext!.size!.width.toString());
                  },
                  flutterWebButtonOptions: FlutterWebButtonOptions(
                    buttonWidth: 200,
                  ),
                ),
              ],
            ),
            Row(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                FlutterWebButton.raiseText(
                  'Raise Text',
                  onPressed: () {},
                  flutterWebButtonOptions: FlutterWebButtonOptions(
                    buttonWidth: 200,
                    buttonRadius: 30,
                  ),
                ),
                FlutterWebButton.textScroll(
                  'Text Scroll',
                  onPressed: () {},
                  flutterWebButtonOptions: FlutterWebButtonOptions(
                    buttonWidth: 200,
                    textColor: const Color(0xFF2A2C2B),
                    buttonBackgroundColor: Colors.transparent,
                    buttonBorderColor: const Color(0XFF2E7D32),
                    buttonRadius: 30,
                  ),
                ),
                FlutterWebButton.simple(
                  'Simple With Hero',
                  onPressed: () {
                    Navigator.of(context).push(MaterialPageRoute(
                        builder: (context) => const HeroPage()));
                  },
                  flutterWebButtonOptions: FlutterWebButtonOptions(
                    buttonWidth: 200,
                    heroTag: 'herotag',
                    isHeroWidget: true,
                  ),
                ),
              ],
            ),
            Row(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                FlutterWebButton.simpleIcon(
                  icon: Icons.menu,
                  onPressed: () {},
                  flutterWebIconOptions:
                      FlutterWebIconOptions(color: Colors.black),
                ),
                FlutterWebButton.socialIcon(
                  flutterWebButtonSocialIcon:
                      FlutterWebButtonSocialIcon.instagram,
                  onPressed: () {},
                  flutterWebIconOptions:
                      FlutterWebIconOptions(color: Colors.pink),
                ),
                FlutterWebButton.socialIconGrow(
                  flutterWebButtonSocialIcon: FlutterWebButtonSocialIcon.github,
                  onPressed: () {},
                  flutterWebIconOptions: FlutterWebIconOptions(),
                ),
                FlutterWebButton.iconGrow(
                  icon: Icons.favorite,
                  onPressed: () {},
                  growAmount: 2,
                  flutterWebIconOptions:
                      FlutterWebIconOptions(color: Colors.red),
                ),
                FlutterWebButton.circle(
                  onPressed: () {},
                  gradient: const LinearGradient(
                      colors: [Colors.pink, Colors.blue],
                      begin: Alignment.centerLeft,
                      end: Alignment.centerRight),
                  child: const Icon(Icons.favorite),
                ),
                FlutterWebButton.widgetButton(
                    onPressed: () {
                      debugPrint('Widget Button Pressed.');
                    },
                    child: const Icon(Icons.widgets))
              ],
            ),
            Row(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                FlutterWebButton.textColorChange(
                  'Text Color Change',
                  onPressed: () {},
                  textAnimatedColor: const Color(0XFF2E7D32),
                  flutterWebButtonOptions: FlutterWebButtonOptions(
                    buttonWidth: 200,
                    eliminateDecoration: true,
                    textColor: const Color(0xFF2A2C2B),
                    fontSize: 24,
                  ),
                ),
                FlutterWebButton.textUnderline(
                  'Underline Animation Default',
                  onPressed: () {},
                  animationDuration: const Duration(milliseconds: 500),
                  flutterTextOptions: FlutterTextOptions(fontSize: 24),
                ),
                FlutterWebButton.textUnderline(
                  'Underline Animation Center',
                  onPressed: () {},
                  flutterTextOptions: FlutterTextOptions(fontSize: 24),
                  animationCrossAxisAlignment: CrossAxisAlignment.center,
                ),
                FlutterWebButton.textMove(
                  'Move Text',
                  onPressed: () {},
                  moveDistanceX: 5,
                  flutterTextOptions: FlutterTextOptions(fontSize: 24),
                )
              ],
            ),
            Row(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                FlutterWebButton.buttonHighlightIconFill(
                    'Icon With Background Fill',
                    icon: Icons.add,
                    onPressed: () {},
                    flutterTextOptions: FlutterTextOptions(
                      fontSize: 20,
                    )),
              ],
            ),
            Row(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                FlutterWebButton.textUnderline(
                  'Underline Animation',
                  onPressed: () {},
                  animationDuration: const Duration(milliseconds: 500),
                  flutterTextOptions: FlutterTextOptions(fontSize: 24),
                ),
              ],
            ),
          ],
        ),
      ),
    );
  }
}

class HeroPage extends StatelessWidget {
  const HeroPage({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: const Text('Hero Page')),
      body: Center(
        child: Hero(
          tag: 'herotag',
          child: Container(
            width: double.infinity,
            height: 400,
            color: Colors.pink,
            child: const Center(
              child: Text(
                'This is a hero!',
                style: TextStyle(fontSize: 50, color: Colors.white),
              ),
            ),
          ),
        ),
      ),
    );
  }
}
